src/unistd.rs

```diff
@@ -2334,3 +2293,99 @@ pub fn getgrgid(gid: Gid) -> Result<Option<Group>> {
         Some(gr) => Ok(Some(Group::from(*gr)))
     }
 }
+
+
+/// This iterator can be used to get all of the users on the system. For example:
+///
+/// ```
+/// use nix::unistd::Users;
+/// eprintln!("Users on this system:\nname\tuid");
+/// Users::new()
+///     .map(
+///         |pw| println!("{}\t{}",
+///                       pw.name.to_string_lossy().into_owned(),
+///                       format!("{}", pw.uid)))
+///     .collect::<Vec<_>>();
+/// 
+/// ```
+#[derive(Debug)]
+#[cfg(not(target_os = "android"))]
+pub struct Users(pub usize);
+
+impl Users {
+    pub fn new() -> Self {
+        unsafe { libc::setpwent(); } 
+        Users(PWGRP_BUFSIZE)
+    }
+
+    pub fn with_bufsize(bufsize: usize) -> Self {
+        unsafe { libc::setpwent(); }
+        Users(bufsize)
+    }
+}
+
+impl Iterator for Users {
+    type Item = Passwd;
+    fn next(&mut self) -> Option<Passwd> {
+        
+        let mut cbuf = vec![0i8; self.0];
+        let mut pwd: libc::passwd =  unsafe { mem::zeroed() };
+        let mut res = ptr::null_mut();
+        
+        let i = unsafe { libc::getpwent_r(&mut pwd, cbuf.as_mut_ptr(), self.0, &mut res) };
+
+        match i {
+            0 if !res.is_null() => {
```

Comment:
You need a way to distinguish between "entry not found" (errno==0) and "not enough buffer space" (errno==ERANGE)